packages/cli/src/utils/input.ts

```diff
@@ -1,14 +1,27 @@
+import BN from 'bignumber.js';
+
 export function parseArgs(args: string): string[] | never {
   if (typeof args !== 'string') throw Error(`Cannot parse ${typeof args}`);
 
   // The following is inspired from ethereum/remix (Copyright (c) 2016)
   // https://github.com/ethereum/remix/blob/89f34fc4f35eca0c386379550c300205d35bfae0/remix-lib/src/execution/txFormat.js#L51-L53
   try {
+     // replace non quoted hex string by quoted hex string
     const MATCH_HEX = /(^|([,[]\s*))(0[xX][0-9a-fA-F]+)/g;
-    args = args.replace(MATCH_HEX, '$2"$3"'); // replace non quoted hex string by quoted hex string
+    args = args.replace(MATCH_HEX, '$2"$3"');
+
+    // replace scientific notation numbers by regular numbers
+    const MATCH_SCIENTIFIC = /(^|([,[]\s*))(\s*[-]?\d+(\.\d+)?e(\+)?\d+\s*)/g;
```

Comment:
Aren't we missing an end-of-string (or `($|([,\]])`) anchor in this regex?

Reply:
I'm not sure I'd call `(^|([,[]\s*))` start of string anchors, they're more like capturing groups. Remember that in this case we are parsing the whole input, not just a single value.

Despite that, if we are missing an "end of string capturing group", then we would need to add them to the other regex-es, since none of them use them.

Reply:
> Despite that, if we are missing an "end of string capturing group", then we would need to add them to the other regex-es, since none of them use them.
Yep, exactly what I was thinking about.

Reply:
Interesting... if we do add the end of string capturing groups in all the regex-es, it would capture ending commas and `]`'s. Take a look: https://regex101.com/r/oES6aY/1/